src/main/java/com/codeborne/selenide/appium/commands/AppiumScrollTo.java

```diff
@@ -0,0 +1,78 @@
+package com.codeborne.selenide.appium.commands;
+
+import com.codeborne.selenide.SelenideElement;
+import com.codeborne.selenide.commands.ScrollTo;
+import com.codeborne.selenide.impl.WebElementSource;
+import io.appium.java_client.AppiumDriver;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.interactions.PointerInput;
+import org.openqa.selenium.interactions.Sequence;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Optional;
+
+import static com.codeborne.selenide.appium.WebdriverUnwrapper.cast;
+import static java.time.Duration.ofMillis;
+import static java.util.Collections.singletonList;
+
+
+@ParametersAreNonnullByDefault
+public class AppiumScrollTo extends ScrollTo {
+
+  private AppiumDriver appiumDriver;
+
+  @Override
+  @Nonnull
+  public SelenideElement execute(SelenideElement proxy, WebElementSource locator, @Nullable Object[] args) {
+    Optional<AppiumDriver> appiumDriverOptional = cast(locator.driver(), AppiumDriver.class);
+    if (!appiumDriverOptional.isPresent()) {
+      return super.execute(proxy, locator, args);
+    }
+
+    this.appiumDriver = appiumDriverOptional.get();
+
+    String previousPageSource = "";
+    while (isElementNotDisplayed(locator) && isNotEndOfPage(previousPageSource)) {
+      previousPageSource = appiumDriver.getPageSource();
+      performScroll();
+    }
+    return proxy;
+  }
+
+  private boolean isElementNotDisplayed(WebElementSource locator){
+    try{
+      return !locator.getWebElement().isDisplayed();
+    }
+    catch (NullPointerException | NoSuchElementException exception){
+      return true;
+    }
+  }
+
+  private boolean isNotEndOfPage(String initialPageSource){
+    return !initialPageSource.equals(appiumDriver.getPageSource());
+  }
+
+  private Dimension getMobileDeviceSize() {
+    return appiumDriver.manage().window().getSize();
+  }
+
+  private void performScroll(){
+    Dimension size = getMobileDeviceSize();
+    PointerInput finger = new PointerInput(PointerInput.Kind.TOUCH, "finger");
+    Sequence sequenceToPerformScroll = getSequenceToPerformScroll(finger, size);
+    appiumDriver.perform(singletonList(sequenceToPerformScroll));
+  }
+
+  private Sequence getSequenceToPerformScroll(PointerInput finger, Dimension size){
```

Comment:
If I have understood correctly, it scrolls only one direction. Such implementation doesn't work for 50% use cases. Am I wrong?

Reply:
Hi @BorisOsipov - Yes it does scroll in one direction. In most of the apps, we only scroll down. Do you suggest implementing something like ScrollOptions with Directions as an enum (UP, DOWN, LEFT, RIGHT)?

In my opinion, not many people would use other ways of scrolling. (YAGNI). Please advise otherwise

Reply:
@amuthansakthivel yes you are right. Scroll UP is more rare case than DOWN. But it is and we have to support it.
Scroll LEFT\RIGHT seems incorrect because it is actually a swipe rather than a scroll.

>Do you suggest implementing something like ScrollOptions with Directions as an enum (UP, DOWN, LEFT, RIGHT)?
>Otherwise, I can create a class called ScrollOptions with timeout and scroll directions.

It is not so easy :). `scrollTo` signature defined in SelenideElement interface and method has no args. For Appium it makes sense to have ScrollOptions arg with Direction\Count of swipes\Etc, but for web it doesn't make sense because it is just js code.

@asolntsev what do you think? how we should handle cases when a command in `Selenide-web` has no args, but we want args for the same command in `selenide-appium`?

We can make a new `SelenideElement scrollTo(ScrollOptions opts);` in Selenide and throw an exception in selenide-web and implement in selenide-appium. But I am not sure that is a good idea.

Reply:
@BorisOsipov - I have also gone through the implementation and understood your point. Is it okay to add a method called scrollUp but that would break other conventions followed across the framework.

I will wait for @asolntsev comments.

Reply:
> @asolntsev what do you think? how we should handle cases when a command in Selenide-web has no args, but we want args for the same command in selenide-appium?

@BorisOsipov Ugh, it's not a simple question. :) 
In this case, cannot we just add parameters "steps", "direction" and "timeout" to Selenide own api? Something like `ScrollOptions`. Aren't these settings also reasonable for web version?

Reply:
@asolntsev - Your attention is needed here

Reply:
@BorisOsipov - In my opinion, we can improvise it later. Not many people would need the option to scroll up.(YAGNI)

Reply:
@amuthansakthivel For all apps I've ever tested we need scroll up for some cases ¯\_(ツ)_/¯. Sure it is not  so often as down scroll.

Also it is unobvious why `$().scroll()To` scrolls only down from user perspective. I will not happy to anwsers such questions in chats\issues when we can fix problem beforehand.

Reply:
@BorisOsipov - Understood and thanks for the details. Let me wait, till @asolntsev comment. I can implement it from my side. Should be easy thing for me anyways.

Reply:
@amuthansakthivel Yes, I agree with @BorisOsipov: when I call `$.scrollTo()`, I expect it to scroll to the element, no matter where it's located. It would creates WTF moment if `$.scrollTo()` scrolls down several times and fails - while my eyes see the element on the screen.

Reply:
@asolntsev - They have to pass the scroll direction. Without that it is not possible to do the search like we do in Selenium. For that we need to have ScrollOptions class to get ScrollDirection as min value and (timeout, number of swipes as optional )

Reply:
@amuthansakthivel Probably we should introduce class `ScrollOptions` in Selenide. 
It seems reasonable to have method `$.scroll(ScrollOptions)` which will encapsulate all possible ways to scroll web browser: using `scrollTo`, `scrollBy`, `scrollIntoView` etc.

Reply:
@asolntsev - 2 points from my side

Users might be already using Selenide.executeJavaScript() method for all these. 
ScrollBy also needs the coordinates. We also need to accommodate that. 

Are you planning to do this now or for the next release? For selenide-appium, I will have a class AppiumScrollOptions with scrollDirection and maxNumberOfSwipeCount as optional inputs

Reply:
@amuthansakthivel I tried to invent class `ScrollOptions` in Selenide, but failed to. 
At the moment, I don't see an easy way to create a generic `ScrollOptions` that would be suitable for different ways of scrolling in web.

I rather suggest to create class `SelenideAppium` - similar to `Selenide`, but with its own static methods. They could also be named `$` and `$$`, but return instances of `SelenideAppiumElement`. This new class will have method `scroll` and other.

Reply:
Hi @asolntsev - Scroll is element specific and it has to be something like $(webelement).scrollTo().click();

If the method scroll is static, then it has to take selenideelement as argument which will break the convention and the code might look something like scroll($(webelement)).click();

Is this what you want? Or do I miss something here?